drivers/adc/adc_mcux_adc16.c

```diff
@@ -180,6 +180,7 @@ static void mcux_adc16_isr(void *arg)
 	SYS_LOG_DBG("Finished channel %d. Result is 0x%04x",
 		    data->channel_id, result);
 
+	data->repeat_buffer = data->buffer;
```

Comment:
This should not be done here but in `adc_context_start_sampling()` instead. When a sampling is repeated, all channels selected for the sequence are sampled again, thus `data->repeat_buffer` should point to the result from the first sampled channel, and here `data->buffer` points to the last one (so this will work only for single channel samplings).

After moving this assignment, [the one from `start_read()`](https://github.com/zephyrproject-rtos/zephyr/blob/5a098d2b8977c7d24d5d56b4c80a8dcc8cf83a47/drivers/adc/adc_mcux_adc16.c#L99) will become redundant.